.github/workflows/ci.yml

```diff
@@ -52,38 +57,87 @@ jobs:
           # Make sure the database is actually working
           psql "${CRATESFYI_DATABASE_URL}"
 
-      - name: Run rustfmt
-        run: cargo fmt -- --check
-
-      - name: Run clippy
-        run: cargo clippy --locked -- -D warnings
-
-      - name: Build docs.rs
+      - name: Build
         run: cargo build --locked
 
-      - name: Test docs.rs
+      - name: Test
         run: cargo test --locked -- --test-threads=1
 
       - name: Clean up the database
         run: docker-compose down --volumes
 
-  docker:
-    name: Docker
+      - name: Clean unused artifacts
+        uses: actions-rs/cargo@v1
+        with:
+          command: sweep
+          args: --file
+
+  fmt:
+    name: Rustfmt
     runs-on: ubuntu-latest
+
     steps:
       - uses: actions/checkout@master
+      - id: install
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          components: rustfmt
+
+      - uses: actions-rs/cargo@v1
         with:
-          fetch-depth: 2
+          command: fmt
+          args: -- --check
 
-      - name: Build the Docker image
-        run: docker build -t docs-rs -f dockerfiles/Dockerfile .
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@master
+      - id: install
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          components: clippy
+      - uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: cargo-sweep
+
+      - name: Cache directories
+        uses: actions/cache@v1
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/bin
+            ~/.cargo/git
+          key: cargo-dirs-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Clippy needs a disambiguator in its cache key, since it finished first it stored its cache and the main build failed to:

```
[warning]reserveCache failed: Cache already exists. Scope: refs/pull/883/merge, Key: cargo-build-c7087fe00-844e47a1d4d07d013bbf2fbfd5751995d510574cded08b4373348486baf5e48c, Version: (null)
```

https://github.com/rust-lang/docs.rs/pull/883/checks?check_run_id=837676712#step:22:2